### PR TITLE
feat(curriculum): add checkout page lab

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

images: https://github.com/freeCodeCamp/cdn/pull/278

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+console.log(els);
+assert.isAtLeast(els.length, 1);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:126]:
```suggestion
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Checkout Page</title>
</head>
<body>

</body>
</html>
```

Comment (Dario-DC) [reply curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:126]: The lab before this has already the boiler in the seed code.

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form input#card-number'));
+assert.exists(secondSection?.querySelector('form input[name="card-number"]'));
+```
+
+All of your input elements that aren't a `type` of `submit` should have a label associated with them.
+
+```js
+const inputs = document.querySelectorAll('input');
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:97]:
```suggestion
const inputs = document.querySelectorAll('input');
assert(inputs.length);
```
It doesn't fail on the seed code

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+const requiredInputs = Array.from(inputs).filter(input => input.required && input.getAttribute('aria-required') === 'true');
+assert.isAtLeast(requiredInputs?.length, 2)
+```
+
+You should use `tabindex` at least once.
+
+```js
+const els = document.querySelectorAll('[tabindex]');
+console.log(els);
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:117]:
```suggestion
```

**Review by Dario-DC — COMMENTED** [review]
It's looking good :+1: I left a couple of comments

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+1. You should have an `h1` with the text `Checkout`.
+1. You should have two `section` elements immediately after the `h1` element.
+1. You should have an `h2` with the text `Your Cart` within the first section.
+1. You should have an image of an item in the first section with appropriate alternate text. You can use this image if you would like: `https://cdn.freecodecamp.org/curriculum/labs/cube.jpg`
+1. You should have an `h2` with the text `Payment Information` within the second section.
+1. You should have a `form` within the second section.
+1. You should have an input with an `id` and `name` of `card-name` within your form and a `label` associated with it.
+1. You should have an input with an `id` and `name` of `card-number` within your form and a `label` associated with it.
+1. At least two of your input elements should be required and have an `aria-required` attribute set to `true`.
+1. You should use `tab-index` at least once.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:24]:
```suggestion
1. You should have two `section` elements immediately after the `h1` element.
1. You should have an `h2` element with the text `Your Cart` within the first section.
1. You should have an image of an item in the first section with appropriate alternate text. You can use this image if you would like: `https://cdn.freecodecamp.org/curriculum/labs/cube.jpg`
1. You should have an `h2` element with the text `Payment Information` within the second section.
1. You should have a `form` element within the second section.
1. You should have an input with an `id` and `name` of `card-name` within your form and a `label` associated with it.
1. You should have an input with an `id` and `name` of `card-number` within your form and a `label` associated with it.
1. At least two of your input elements should be required and have an `aria-required` attribute set to `true`.
1. You should use `tab-index` at least once.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+# --hints--
+
+You should have an `h1` with the text `Checkout`.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:28]:
```suggestion
You should have an `h1` element with the text `Checkout`.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+```js
+const h1 = document.querySelector('h1');
+const firstSibling = h1?.nextElementSibling;
+const secondSibling = firstSibling?.nextElementSibling;
+assert.equal(firstSibling?.tagName, 'SECTION');
+assert.equal(secondSibling?.tagName, 'SECTION');
+```
+
+You should have an `h2` with the text `Your Cart` within the first section.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:50]:
```suggestion
You should have an `h2` element with the text `Your Cart` within the first section.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+You should have at least one image with alternate text in your first section.
+
+```js
+const firstSection = document.querySelectorAll('section')?.[0];
+assert.isAtLeast(firstSection?.querySelector('img')?.alt?.length, 1);
+```
+
+You should have an `h2` with the text `Payment Information` within the second section.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:64]:
```suggestion
You should have an `h2` element with the text `Payment Information` within the second section.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.equal(secondSection?.querySelector('h2')?.innerText, 'Payment Information');
+```
+
+You should have a `form` within the second section.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:71]:
```suggestion
You should have a `form` element within the second `section` element.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form'));
+```
+
+You should have an input with an `id` and `name` of `card-name` within your form.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:78]:
```suggestion
You should have an `input` element with an `id` and `name` of `card-name` within your form.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form input#card-name'));
+assert.exists(secondSection?.querySelector('form input[name="card-name"]'));
+```
+
+You should have an input with an `id` and `name` of `card-number` within your form.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:86]:
```suggestion
You should have an `input` element with an `id` and `name` of `card-number` within your form.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form input#card-number'));
+assert.exists(secondSection?.querySelector('form input[name="card-number"]'));
+```
+
+All of your input elements that aren't a `type` of `submit` should have a label associated with them.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:94]:
```suggestion
All of your `input` elements that aren't a `type` of `submit` should have a `label` element associated with them.
```

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+```js
+const inputs = document.querySelectorAll('input');
+inputs.forEach(input => {
+  if (input.type !== 'submit') {
+    assert.exists(document.querySelector(`label[for="${input.id}"]`))
+  }
+});
+```
+
+At least two of your input elements should be required and have an `aria-required` attribute set to `true`.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:105]:
```suggestion
At least two of your `input` elements should be required and have an `aria-required` attribute set to `true`.
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
I think using more the word element in conjuction with element names can make things clearer (and easier to translate)

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+
+You should have at least two `section` elements after the `h1` element.
+
+```js
+const h1 = document.querySelector('h1');
+const firstSibling = h1?.nextElementSibling;
+const secondSibling = firstSibling?.nextElementSibling;
+assert.equal(firstSibling?.tagName, 'SECTION');
+assert.equal(secondSibling?.tagName, 'SECTION');
+```
```
[the remark this answers is not in the document]
Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:48]: Thanks. Just gonna leave this one how it is I think.

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md (hunk 2 above)
Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md]: ✅

File: curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md (hunk 1 above)
Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md:133]: ✅

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**